integrationtests/gitjob/controller/controller_test.go

```diff
@@ -226,6 +251,110 @@ var _ = Describe("GitJob controller", func() {
 			}).Should(BeTrue())
 		})
 	})
+
+	When("creating a gitRepo that references a nonexistent helm secret", func() {
+		var (
+			gitRepo                v1alpha1.GitRepo
+			gitRepoName            string
+			helmSecretNameForPaths string
+			helmSecretName         string
+		)
+
+		JustBeforeEach(func() {
+			gitRepoName = "test-no-for-paths-secret"
+			gitRepo = createGitRepo(gitRepoName)
+			gitRepo.Spec.HelmSecretNameForPaths = helmSecretNameForPaths
+			gitRepo.Spec.HelmSecretName = helmSecretName
+			// Create should return an error
+			err := k8sClient.Create(ctx, &gitRepo)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(simulateGitPollerUpdatingCommitInStatus(gitRepo, commit)).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(ctx, &gitRepo)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Context("helmSecretNameForPaths secret does not exist", func() {
+			BeforeEach(func() {
+				helmSecretNameForPaths = "secret-does-not-exist"
+				helmSecretName = ""
+			})
+			It("logs an error about HelmSecretNameForPaths not being found", func() {
+				Eventually(func() bool {
+					strLogs := logsBuffer.String()
+					return strings.Contains(strLogs, `failed to look up HelmSecretNameForPaths, error: Secret \"secret-does-not-exist\" not found`)
+				}).Should(BeTrue())
+			})
+
+			It("doesn't create RBAC resources", func() {
+				Consistently(func() bool {
+					saName := name.SafeConcatName("git", gitRepo.Name)
+					ns := types.NamespacedName{Name: saName, Namespace: gitRepo.Namespace}
+
+					if err := k8sClient.Get(ctx, ns, &corev1.ServiceAccount{}); !errors.IsNotFound(err) {
+						return false
+					}
+					if err := k8sClient.Get(ctx, ns, &rbacv1.Role{}); !errors.IsNotFound(err) {
+						return false
+					}
+					if err := k8sClient.Get(ctx, ns, &rbacv1.RoleBinding{}); !errors.IsNotFound(err) {
+						return false
+					}
+					return true
+				}, time.Second*5, time.Second*1).Should(BeTrue())
+			})
+
+			It("doesn't create the job", func() {
+				Consistently(func() bool {
+					jobName := name.SafeConcatName(gitRepoName, name.Hex(repo+commit, 5))
+					newJob := &batchv1.Job{}
+					err := k8sClient.Get(ctx, types.NamespacedName{Name: jobName, Namespace: gitRepo.Namespace}, newJob)
+					return errors.IsNotFound(err)
+				}, time.Second*5, time.Second*1).Should(BeTrue())
+			})
+		})
+		Context("helmSecretName secret does not exist", func() {
+			BeforeEach(func() {
+				helmSecretNameForPaths = ""
+				helmSecretName = "secret-does-not-exist"
+			})
+			It("logs an error about HelmSecretNameForPaths not being found", func() {
```

Comment:
Shouldn't this be:
```suggestion
			It("logs an error about HelmSecretName not being found", func() {
```
?

Reply:
Thanks for catching this copy/paste evil thing!

The test was passing because that was the previous value checked and was still in the logs buffer. 
I fixed the name and error message check AND I'm cleaning the `logsBuffer` in the `AfterEach` statement for those tests so the logs are clean and we don't get entries from previous tests.